### PR TITLE
PODS-2711: Create generic controller and view for CRN

### PR DESCRIPTION
**Pull request by chris-ianson**
- Created a base controller for CompanyRegistrationNumber
- Added new CompanyRegistrationNumberUpdateController that only shows a text field
- Added a new companyRegistrationNumberUpdate view that only contains a text field
- Refactered CompanyRegistrationNumberController for trustees to it can use base controller

### CODE REVIEW

File: app/controllers/register/establishers/company/CompanyRegistrationNumberUpdateController.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+class CompanyRegistrationNumberUpdateController @Inject()(
+                                                     val appConfig: FrontendAppConfig,
+                                                     override val messagesApi: MessagesApi,
+                                                     userAnswersService: UserAnswersService,
+                                                     @EstablishersCompany navigator: Navigator,
+                                                     authenticate: AuthAction,
+                                                     getData: DataRetrievalAction,
+                                                     allowAccess: AllowAccessActionProvider,
+                                                     requireData: DataRequiredAction,
+                                                     formProvider: CompanyRegistrationNumberFormProvider
```
Comment (grantrobson) [inline app/controllers/register/establishers/company/CompanyRegistrationNumberUpdateController.scala]: Will this work for variations, where there will be no boolean, i.e. hasCrn?

File: app/controllers/register/CompanyRegistrationNumberBaseController.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  def id(index: Index): TypedIdentifier[CompanyRegistrationNumber]
+
+  def onPageLoad(mode: Mode, srn: Option[String], index: Index): Action[AnyContent] =
+    (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData).async {
+      implicit request =>
+        val redirectResult = request.userAnswers.get(id(index)) match {
+          case None =>
+            Ok(addView(mode, index, srn))
+          case Some(value) =>
```
Comment (madhuranakate) [inline app/controllers/register/CompanyRegistrationNumberBaseController.scala]: We shouldn't be populating the value if user's previous answer was CRN.No(reason)

Comment (chris-ianson) [reply app/controllers/register/CompanyRegistrationNumberBaseController.scala]: If the users answer was they didn't have a CRN, they shouldn't be shown the page to update/add it. So the user would never be shown the CompanyRegistrationNumberUpdateController page.

File: app/controllers/register/establishers/company/CompanyRegistrationNumberUpdateController.scala (hunk 1 above)
Comment (chris-ianson) [reply app/controllers/register/establishers/company/CompanyRegistrationNumberUpdateController.scala]: I want to test this in the other related story

**Pull request closed without merging**